src/rdmurmur2.h

```diff
@@ -0,0 +1,30 @@
+#ifndef __RDMURMUR2___H__
+#define __RDMURMUR2___H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "MurMurHash2.h"
+
+typedef uint32_t rd_murmur2_t;
+
+#define MURMUR2_SEED 0x9747b28c
+
+#define INT32_TO_UINT32 0x7FFFFFFF
+
+static RD_INLINE rd_murmur2_t rd_murmur2(const char *data, size_t data_len) {
+  /*
+   * last bit is set to 0 because the java implementation uses int_32
```

Comment:
Isn't a cast to int32_t sufficient?

Reply:
no since the java version flips the first bit to make it positive and we need to maintain that behaviour